tools/changelog/changelog/scope.go

```diff
@@ -0,0 +1,88 @@
+package changelog
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Structured scope annotation, with YAML marshaling to and from a string.
+type Scope struct {
+	Primary   string
+	SubScopes []string
+}
+
+func (s Scope) String() string {
+	if len(s.SubScopes) == 0 {
+		return s.Primary
+	}
+	return s.Primary + "/" + strings.Join(s.SubScopes, ",")
+}
+
+func (s Scope) MarshalYAML() (any, error) {
+	return s.String(), nil
+}
+
+func (s *Scope) UnmarshalYAML(unmarshal func(any) error) error {
+	var value string
+	if err := unmarshal(&value); err != nil {
+		return err
+	}
+
+	primary, subscopes, err := ParseScope(Config{}, value, true)
+	if err != nil {
+		return err
+	}
+	s.Primary = primary
+	s.SubScopes = subscopes
+
+	return nil
+}
+
+// ParseScope ensures a scope string is in the permitted list and if so, returns the scope and permitted subscopes
```

Comment:
Am I understanding the intent correctly?

```suggestion
// ParseScope ensures a scope string is in the list permitted by Config and if so, returns the scope and permitted subscopes
```